selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+    stacks.waitToolbarTitleName();
+    seleniumWebDriver.navigate().back();
+    newWorkspace.waitPageLoad();
+  }
+
+  @Test
+  public void checkSearchField() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.typeToSearchInput("Java");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.typeToSearchInput("java");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.typeToSearchInput("JAVA");
+    newWorkspace.waitStacks(EXPECTED_JAVA_STACKS);
+    newWorkspace.typeToSearchInput("");
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkRamSelection() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.selectStack(JAVA);
+    newWorkspace.waitStackSelected(JAVA);
+    newWorkspace.waitRamValue(MACHINE_NAME, 2.0);
+    newWorkspace.typeToRamField("");
+    newWorkspace.waitRedRamFieldBorders();
+    newWorkspace.waitTopCreateWorkspaceButtonDisabled();
+    newWorkspace.waitBottomCreateWorkspaceButtonDisabled();
+
+    newWorkspace.typeToRamField(Double.toString(MAX_RAM_VALUE));
+    newWorkspace.waitRedRamFieldBordersDisappearance();
+    newWorkspace.waitTopCreateWorkspaceButtonEnabled();
+    newWorkspace.waitBottomCreateWorkspaceButtonEnabled();
+
+    newWorkspace.clickOnIncrementMemoryButton(MACHINE_NAME);
+    newWorkspace.waitRamValue(MACHINE_NAME, MAX_RAM_VALUE);
+
+    newWorkspace.clickOnDecrementMemoryButton(MACHINE_NAME);
+    newWorkspace.waitRamValue(MACHINE_NAME, MAX_RAM_VALUE - RAM_CHANGE_STEP);
+
+    newWorkspace.typeToRamField("");
+    newWorkspace.waitRedRamFieldBorders();
+    newWorkspace.waitTopCreateWorkspaceButtonDisabled();
+    newWorkspace.waitBottomCreateWorkspaceButtonDisabled();
+
+    newWorkspace.typeToRamField(Double.toString(MIN_RAM_VALUE));
+    newWorkspace.waitRedRamFieldBordersDisappearance();
+    newWorkspace.waitTopCreateWorkspaceButtonEnabled();
+    newWorkspace.waitBottomCreateWorkspaceButtonEnabled();
+    newWorkspace.clickOnDecrementMemoryButton(MACHINE_NAME);
+    newWorkspace.waitRamValue(MACHINE_NAME, MIN_RAM_VALUE);
+    newWorkspace.clickOnIncrementMemoryButton(MACHINE_NAME);
+    newWorkspace.waitRamValue(MACHINE_NAME, MIN_RAM_VALUE + RAM_CHANGE_STEP);
+
+    newWorkspace.clickAndHoldIncrementMemoryButton(MACHINE_NAME, 3);
+    newWorkspace.waitRamValueInSpecifiedRange(MACHINE_NAME, 3, MAX_RAM_VALUE);
+
+    double currentRamAmount = newWorkspace.getRAM(MACHINE_NAME);
+    newWorkspace.clickAndHoldDecrementMemoryButton(MACHINE_NAME, 3);
+    newWorkspace.waitRamValueInSpecifiedRange(MACHINE_NAME, MIN_RAM_VALUE, currentRamAmount - 2);
+  }
+
+  private void checkNotValidNames() {
+    NOT_VALID_NAMES.forEach(
+        name -> {
+          newWorkspace.typeWorkspaceName("temporary");
+
```

Comment:
it's better not to divide action and verification sub-steps by empty row to make separate steps more obvious.